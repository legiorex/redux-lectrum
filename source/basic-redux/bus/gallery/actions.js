// Types
import { SHOW_NEXT_PHOTO, SHOW_PREV_PHOTO, SHOW_SELECTED_PHOTO } from "./types";

export const showNextPhoto = () => {
    return {
        type: SHOW_NEXT_PHOTO,
    };

};

export const showPrevPhoto = () => {
    return {
        type: SHOW_PREV_PHOTO,
    };

};

export const showSelectedPhoto = (photoIndex) => {
    return {
        type:    SHOW_SELECTED_PHOTO,
        payload: photoIndex,
    };

};
