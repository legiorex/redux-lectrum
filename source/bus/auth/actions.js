// Types
import { types } from './types';

export const authActions = {
    // Sync
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },

    // Async
    signupAsync: (userData) => {
        return {
            type:    types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (userLogin) => {
        return {
            type:    types.LOGIN_ASYNC,
            payload: userLogin,
        };
    },
};
