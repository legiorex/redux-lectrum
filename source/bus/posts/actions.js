// Types
import { FILL_POSTS, FETCH_POST_ASYNC, CREATE_POST_ASYNC } from "./types";
import { api } from "../../REST";

// Instruments

export const fillPosts = (posts) => {
    return {
        type:    FILL_POSTS,
        payload: posts,
    };
};

export const fetchPostsAsync = () => async (dispatch) => {
    dispatch({
        type: FETCH_POST_ASYNC,
    });

    const response = await api.posts.fetch();
    const result = await response.json();

    dispatch(fillPosts(result.data));
};
export const createPostAsync = (comment) => async (dispatch) => {

    dispatch({
        type: CREATE_POST_ASYNC,
    });

    const response = await api.posts.create(comment);    
    const result = await response.json();

    console.log(result.data);
    
    dispatch(fillPosts(result.data));
};
