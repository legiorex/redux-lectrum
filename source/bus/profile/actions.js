// Types
import { types } from './types';

export const profileActions = {
    fillProfile: (profile) => {
        return {
            type:    types.FILL_PROFILE,
            payload: profile,
        };
    },
};
