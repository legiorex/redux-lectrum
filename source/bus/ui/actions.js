// Types
import { types } from './types';

export const uiActions = {
    startFetching: () => {
        return {
            type: types.START_FETCHING,
        };
    },
    stopFetching: () => {
        return {
            type: types.STOP_FETCHING,
        };
    },
    emitError: (error, meta = null) => {
        return {
            type:    types.EMIT_ERROR,
            payload: error,
            error:   true,
            meta,
        };
    },
};
