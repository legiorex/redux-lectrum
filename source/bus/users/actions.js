// Types
import { types } from './types';

export const usersActions = {
    // Sync
    fillUsers: (users) => {
        return {
            type:    types.FILL_USERS,
            payload: users,
        };
    },
    clearUsers: () => {
        return {
            type: types.CLEAR_USERS,
        };
    },

    // Async
    fetchUsersAsync: () => {
        return {
            type: types.FETCH_USERS_ASYNC,
        };
    },
};
