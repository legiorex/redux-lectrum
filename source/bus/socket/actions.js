// Actions
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postsActions } from '../posts/actions';

export const socketActions = {
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState());
        });
        socket.on('disconnect', () => {
            dispatch(uiActions.setOfflineState());
        });
    },
    listenPosts: () => (dispatch, getState) => {
        socket.on("create", (event) => {
            const { data: post } = JSON.parse(event);

            dispatch(postsActions.createPost(post));
        });
        socket.on("remove", (event) => {

            const { data: post } = JSON.parse(event);

            dispatch(postsActions.removePost(post));
        });
        socket.on("like", (event) => {

            const { data, meta } = JSON.parse(event);

            if (meta.action === 'like') {
                const liker = getState().users.find((user) => user.get('id') === data.userId).delete('avatar');

                dispatch(postsActions.likePost({
                    postId: data.postId,
                    liker,
                }));
            } else {
                dispatch(postsActions.unlikePost(data));
            }

        });
    },
};
