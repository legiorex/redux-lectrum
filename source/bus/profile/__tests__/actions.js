// Actions
import { profileActions } from "../actions";

describe('profile actions:', () => {

    test("fillProfile", () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
    });
    test("updateAvatar", () => {
        expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
    });
    test("clearProfile", () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });

    test("updateNameAsync", () => {
        expect(profileActions.updateNameAsync(__.newName)).toMatchSnapshot();
    });
    test("updateAvatarAsync", () => {
        expect(
            profileActions.updateAvatarAsync(__.newAvatar)
        ).toMatchSnapshot();
    });
    test("updatePasswordAsync", () => {
        expect(
            profileActions.updatePasswordAsync(__.newPassword)
        ).toMatchSnapshot();
    });
});
