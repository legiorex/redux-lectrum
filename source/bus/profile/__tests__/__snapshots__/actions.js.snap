// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions: clearProfile 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions: fillProfile 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions: updateAvatar 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`profile actions: updateAvatarAsync 1`] = `
Object {
  "payload": Array [
    "avatar",
  ],
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`profile actions: updateNameAsync 1`] = `
Object {
  "payload": Object {
    "firstName": "Walter",
    "lastName": "White",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`profile actions: updatePasswordAsync 1`] = `
Object {
  "payload": Object {
    "newPassword": 123456,
    "oldPassword": 12345,
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
